sync_diff_inspector/source/source_test.go

```diff
@@ -685,6 +685,106 @@ func TestSource(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestRouterRules(t *testing.T) {
+	host, isExist := os.LookupEnv("MYSQL_HOST")
+	if host == "" || !isExist {
+		return
+	}
+	portStr, isExist := os.LookupEnv("MYSQL_PORT")
+	if portStr == "" || !isExist {
+		//return
+	}
+	port, err := strconv.Atoi(portStr)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	r, err := router.NewTableRouter(false, []*router.TableRule{
+		// make sure this rule works
+		{
+			SchemaPattern: "schema1",
+			TablePattern:  "tbl",
+			TargetSchema:  "schema2",
+			TargetTable:   "tbl",
+		},
+	})
+	cfg := &config.Config{
+		LogLevel:         "debug",
+		CheckThreadCount: 4,
+		ExportFixSQL:     true,
+		CheckStructOnly:  false,
+		DataSources: map[string]*config.DataSource{
+			"mysql1": {
+				Host: host,
+				Port: port,
+				User: "root",
+			},
+			"tidb": {
+				Host: host,
+				Port: port,
+				User: "root",
+			},
+		},
+		Routes: nil,
+		Task: config.TaskConfig{
+			Source:      []string{"mysql1"},
+			Routes:      nil,
+			Target:      "tidb",
+			CheckTables: []string{"schema2.tbl"},
+			OutputDir:   "./output",
+			SourceInstances: []*config.DataSource{
+				{
+					Host:           host,
+					Port:           port,
+					User:           "root",
+					Router:         r,
+					RouteTargetSet: make(map[string]struct{}),
+				},
+			},
+			TargetInstance: &config.DataSource{
+				Host: host,
+				Port: port,
+				User: "root",
+			},
+			TargetCheckTables: nil,
+			FixDir:            "output/fix-on-tidb0",
+			CheckpointDir:     "output/checkpoint",
+			HashFile:          "",
+		},
+		ConfigFile:   "config.toml",
+		PrintVersion: false,
+	}
+	cfg.Task.TargetCheckTables, err = filter.Parse([]string{"schema2.tbl"})
+	require.NoError(t, err)
+	cfg.Task.SourceInstances[0].RouteTargetSet[dbutil.TableName("schema2", "tbl")] = struct{}{}
+
+	// create table
+	conn, err := sql.Open("mysql", fmt.Sprintf("root:@tcp(%s:%d)/?charset=utf8mb4", host, port))
+	require.NoError(t, err)
+
+	conn.Exec("CREATE DATABASE IF NOT EXISTS schema1")
+	conn.Exec("CREATE TABLE IF NOT EXISTS `schema1`.`tbl` (`a` int, `b` varchar(24), `c` float, `d` datetime, primary key(`a`, `b`))")
```

Comment:
Do we need to drop this table/schema first?

Reply:
we need keep this `schema1`.`tbl` for next router check.